nebula3/gclient/net/Connection.py

```diff
@@ -171,6 +171,8 @@ def authenticate(self, user_name, password):
         try:
             resp = self._connection.authenticate(user_name, password)
             if resp.error_code != ErrorCode.SUCCEEDED:
+                self._connection.is_used = False
+                self._connection = None
```

Comment:
Authenticate failed does not mean the connection is broken, Why `set _connection = None`, we can just set `_connection.is_used` as `False`.

Let's see the connection pool how to call `authenticate` function:
```
    def get_session(self, user_name, password, retry_connect=True):
        """get session

        :param user_name: the user name to authenticate graphd
        :param password: the password to authenticate graphd
        :param retry_connect:
        :return: Session
        """
        connection = self.get_connection()
        if connection is None:
            raise NotValidConnectionException()
        try:
            auth_result = connection.authenticate(user_name, password)
            return Session(connection, auth_result, self, retry_connect)
        except Exception:
            raise
```
the auth operation is undependent with connection, so auth failed does not Influence the `connection` unless there are IO Error.

Reply:
I refer to the code under `release` function of `session.py`. I think you may right,  `self._connection.is_used = False` is  enought here

Reply:
Done as agreed.